### PR TITLE
Remove `TenantConverterRegistry`

### PR DESCRIPTION
**Pull request by armiol**
The `DatastoreStorageFactory` configuration allows supplying a custom `NamespaceConverter`. It is used to put the `TenantId` values into accordance with the Datastore namespaces, in which the tenant-specific data resides.

Before this PR we had a singleton called `TenantConverterRegistry`, which could store several converters for each of the Datastore Project IDs used in the application. 

As of now, the `DatastoreStorageFactory` is, in fact, a property of `ServerEnvironment`. So there is the only instance of this factory per JVM. Therefore, it's impossible to configure it with multiple Datastore Project IDs. 

This PR removes the redundant singleton-registry of converters and propagates the one set by the framework user (or a framework default converter, if no custom set) down to the routines working with `Namespace`s.

### CODE REVIEW

**Comment by armiol** [comment]
@dmdashenkov PTAL.

File: datastore/src/main/java/io/spine/server/storage/datastore/DatastoreStorageFactory.java [hunk 1]
```diff
@@ -182,18 +184,12 @@ protected DsPropertyStorage createPropertyStorage(ContextSpec spec) {
     }
 
     private NamespaceSupplier createNamespaceSupplier(boolean multitenant) {
-        @Nullable String defaultNamespace;
         if (multitenant) {
             checkHasNoNamespace(datastore);
-            defaultNamespace = null;
+            return NamespaceSupplier.singleTenant();
```
Comment (dmdashenkov) [inline datastore/src/main/java/io/spine/server/storage/datastore/DatastoreStorageFactory.java]: This surely must be a typo: if `multitenant`, return single tenant supplier.

File: datastore/src/main/java/io/spine/server/storage/datastore/tenant/DefaultNamespaceConverter.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    @Override
+    protected String toString(TenantId tenantId) {
+        return forTenant(tenantId).toString(tenantId);
+    }
+
+    @Override
+    protected TenantId toTenantId(String namespace) {
+
```
Comment (dmdashenkov) [inline datastore/src/main/java/io/spine/server/storage/datastore/tenant/DefaultNamespaceConverter.java]: Please remove this line.

**Review by dmdashenkov — CHANGES_REQUESTED** [review]
@armiol, please see my comments.

File: datastore/src/main/java/io/spine/server/storage/datastore/DatastoreStorageFactory.java [hunk 3]
```diff
@@ -186,9 +186,11 @@ protected DsPropertyStorage createPropertyStorage(ContextSpec spec) {
     private NamespaceSupplier createNamespaceSupplier(boolean multitenant) {
         if (multitenant) {
             checkHasNoNamespace(datastore);
-            return NamespaceSupplier.singleTenant();
-        } else {
             return NamespaceSupplier.multitenant(converterFactory);
+        } else {
+            String defaultNamespace = datastore.getOptions()
+                                               .getNamespace();
+            return NamespaceSupplier.singleTenant(defaultNamespace == null ? "" : defaultNamespace);
```
Comment (dmdashenkov) [inline datastore/src/main/java/io/spine/server/storage/datastore/DatastoreStorageFactory.java]: Please use `Strings.nullToEmpty`.

**Review by dmdashenkov — APPROVED** [review]

**Pull request merged**